docs/source/guide/pt-1-intro.md

```diff
@@ -50,56 +52,91 @@ the circuit on a noisy simulator, and returns the probability of the ground
 expectation values.
 
 ```{code-cell} ipython3
-# Compute the expectation value of the |0><0| observable.
-noisy_value = execute(circuit)
+# Set the intensity of the noise
+NOISE_LEVEL = 0.1
+
+# Compute the expectation value of the |0><0| observable
+# in both the noiseless and the noisy setup
 ideal_value = execute(circuit, noise_level=0.0)
-print(f"Error without mitigation: {abs(ideal_value - noisy_value) :.3}")
+noisy_value = execute(circuit, noise_level=NOISE_LEVEL)
+
+print(f"Error without twirling: {abs(ideal_value - noisy_value) :.3}")
 ```
 
 ## Apply PT
-Pauli Twirling can be easily implemented with the function
-{func}`.pauli_twirl_circuit()`.
+PT can be applied by first generating twirled variants of the circuit with the function
+{func}`.generate_pauli_twirl_variants` from the `mitiq.pt` module, 
+and then averaging over the results obtained by executing those variants.
 
-```{code-cell} ipython3
-from mitiq import pt
-mitigated_result = pt.pauli_twirl_circuit(
-    circuit=circuit,
-)
-```
+The more variants are generated and averaged over, the more visible the results of PT are.
+In this example we generate 5 twirled variants of the circuit, by setting the  `num_circuits` 
+argument of the function {func}`.generate_pauli_twirl_variants` (default value is 10.)
 
 ```{code-cell} ipython3
-# print(f"Error with mitigation (PT): {abs(ideal_value - mitigated_result) :.3}")
+from functools import partial
+import numpy as np
+from mitiq.executor.executor import Executor
+from mitiq.pt import generate_pauli_twirl_variants
+
+# Generate twirled circuits
+NUM_TWIRLED_VARIANTS = 5
+twirled_circuits = generate_pauli_twirl_variants(circuit, num_circuits=NUM_TWIRLED_VARIANTS)
+# Average results executed over twirled circuits
+pt_vals = Executor(partial(execute, noise_level=NOISE_LEVEL)).evaluate(twirled_circuits)
+mitigated_result = np.average(pt_vals)
+
+print(f"Error with twirling: {abs(ideal_value - mitigated_result) :.3}")
 ```
 
-Here we observe that the application of PT does not reduce the estimation error when compared
-to the unmitigated result. The intended effect was to only tailor the noise. 
+The idea behind Pauli Twirling is that it leaves the effective logical circuit unchanged, while tailoring the noise into stochastic Pauli errors.
 
 ```{admonition} Note:
-PT is designed to transform the noise simulated in this example,
-but it should not be expected to always be a positive effect.
-In this sense, it is more of a noise tailoring technique, designed
-to be composed with other techniques rather than an error mitigation
-technique in and of itself.
+Pauli Twirling is designed to transform noise, such as the coherent noise simulated in the example above,
+but it should not be expected to always have a positive effect. In this sense, it is more of a noise tailoring technique, 
+designed to be composed with other techniques rather than an error mitigation technique in itself.
```

Comment:
Can we add a note here that one goal of PT is to make other techniques more effective when stacked together?